src/db/mongodb/mod.rs

```diff
@@ -56,8 +56,10 @@ impl MongoDb {
     }
 
     /// Creates a collection if it does not exist.
-    pub async fn create_collection<T: MongoDbCollection>(&self) {
+    pub async fn create_indexes<T: MongoDbCollection + Send + Sync>(&self) -> Result<(), Error> {
```

Comment:
At this point `initialize` might be a better name?

Reply:
yea, might be ... `initialize` is a bit opaque though and according to the docs `self.db.create_collection` how we call it (without any options) is pretty much a no-op (as MongoDb creates collections lazily when you start inserting). I think @Alexandcoats still thinks it's a necessary call so I kept it, but the true purpose of this call is in fact to create the indexes.

Reply:
However ... i don't care so much, I just wanted to keep the current "naming convention".

Reply:
It's important to call `create_collection` because otherwise the indexes are not created at start up. I also don't mind the current name, was just a suggestion.

Reply:
ah, I thought that might have been the reason ... so in that case we might want to still be able to call it (`self.db.create_collection`) separate from creating the associated indexes :thinking: and controlling this via a boolean argument.

Reply:
This is not what I had in mind when I suggested moving this functionality here. I meant that the create_indexes fn in the trait should also create the collection for simplicity.

Reply:
Not sure I understand what you want. The `create_indexes` fn in the trait is supposed to be overwritten. :thinking:

Reply:
the collection is already created in `crate_indexes` of the `MongoDb` type before it calls the trait method. So isn't that simple enough? @Alexandcoats

Reply:
I'm saying we shouldn't have `MongoDb::create_indexes`, we should have `MongoDb::create_collection` which is called by the trait's `create_indexes`.

Reply:
alright, we can do that.